File: src/ch03/code_3.py
from typing import List

from rich import print


def search_min_val(nums: List[int]) -> int:
    min_val: int = nums[0]
    for num in nums:
        if num < min_val:
            min_val = num

    return min_val


if __name__ == '__main__':
    nums: List[int] = [3, 4, 1, 2, 5, 7]
    result: int = search_min_val(nums=nums)
    print(result)

File: src/ch03/code_3.rs
fn search_min_val(nums: Vec<i32>) -> i32 {
    let mut min_val: i32 = nums[0];
    for num in nums {
        if num < min_val {
            min_val = num;
        }
    }
    min_val
}


fn main() {
    let nums: Vec<i32> = vec![3, 4, 1, 2, 5, 7];
    let result: i32 = search_min_val(nums);

    println!("{}", result);
}